pattern_generator.rb

```diff
@@ -41,39 +53,35 @@ def create_enumerated_array(pattern)
   end
 
   def iterations_per_enumerator(iteration)
-    @imap = []
-    ia = @pattern_arr.reverse.map{|item| item.is_a?(Array) ? item.count : item}
+    ia = @pattern_arr.reverse.map{ |item| item.is_a?(Array) ? item.count : item }
     
     ia.each do |item|
       if item.is_a?(String)
-        @imap << 'lit'
+        imap << 'lit'
         next
       elsif iteration == 0
-        @imap << nil
+        imap << nil
         next
       elsif iteration < item
-        @imap << iteration
+        imap << iteration
       else
-        @imap << iteration % item
+        imap << iteration % item
       end
       iteration = iteration / item
     end
-    @imap.reverse!
+    imap.reverse!
+  end
+
+  def pattern_logic(x)
```

Comment:
Maybe call this `generator_pattern_logic(x)` that way we know if will go in generate or something

Up to you!